nearswap/tests/fee_simulation.rs

```diff
@@ -0,0 +1,234 @@
+    )
+    .assert_success();
+
+    // register and add deposit to accounts
+    register_deposit_acc(&nearswap, &lp1, to_yocto("35"));
+    register_deposit_acc(&nearswap, &lp2, to_yocto("35"));
+
+    // Add to accounts whitelist
+    call!(
+        lp1,
+        nearswap.add_to_account_whitelist(&vec![to_va(dai())])
+    )
+    .assert_success();
+    call!(
+        lp2,
+        nearswap.add_to_account_whitelist(&vec![to_va(dai())])
+    )
+    .assert_success();
+
+    // Depositing tokens in account deposit
+    call!(
+        lp1,
+        token1.ft_transfer_call(to_va(clp_contract()), to_yocto("100").into(), None, "".to_string()),
+        deposit = 1
+    )
+    .assert_success();
+    call!(
+        lp2,
+        token1.ft_transfer_call(to_va(clp_contract()), to_yocto("100").into(), None, "".to_string()),
+        deposit = 1
+    )
+    .assert_success();
+
+    print!("Token deposited in contract account deposit");
+
+    // Adding liquidity
+    let lp1_shares = call!(
+        lp1,
+        nearswap.add_liquidity(dai(), to_yocto_str("9"), to_yocto_str("90"), U128(0)),
+        deposit = 1
+    )
+    .unwrap_json::<U128>();
+    let lp2_shares = call!(
+        lp2,
+        nearswap.add_liquidity(dai(), to_yocto_str("1"), to_yocto_str("10.01"), U128(0)),
+        deposit = 1
+    )
+    .unwrap_json::<U128>();
+
+    print!("Added Liquidity!");
+    register_deposit_acc(&nearswap, &alice, to_yocto("25"));
+    call!(
+        alice,
+        nearswap.add_to_account_whitelist(&vec![to_va(dai())])
+    )
+    .assert_success();
+    print!("Alice Registered!");
+    // Alice buy token by paying 5 NEAR
+    let mut before_swap_token = view!(
+        nearswap.get_deposit_token(alice.account_id.clone(), dai())
+    ).unwrap_json::<U128>();
+    assert_close(before_swap_token, to_yocto("0"), 0);
+
+    let expected_receive = view!(
+            nearswap.price_near_to_token_in(dai(), to_yocto_str("5"))
+        ).unwrap_json::<U128>();
+
+    let mut pool_before = view!(nearswap.pool_info(&dai())).unwrap_json::<PoolInfo>();
+
+    call!(
+        alice,
+        nearswap.swap_near_to_token_exact_in(to_yocto_str("5"), dai(), expected_receive),
+        deposit = 1
+    ).assert_success();
+
+    let mut after_swap_token = view!(
+            nearswap.get_deposit_token(alice.account_id, dai())
+        ).unwrap_json::<U128>();
+
+    assert_eq!(
+        to_u128(before_swap_token) + to_u128(expected_receive),
+        to_u128(after_swap_token), "Near to token swap unsuccessful");
+
+    // Check if fee is deducted - Near deposited into pool must be 0.997 * near amount
+    // Fee - 0.3%
+    let out = mock_calc_out(
+        to_yocto("5")*997/1000,
+        to_u128(pool_before.ynear), to_u128(pool_before.tokens)
+    );
+    assert_eq!(out, to_u128(after_swap_token), "Wrong amount of fee deducted");
+
+    pool_before = view!(nearswap.pool_info(&dai())).unwrap_json::<PoolInfo>();
+    let before_swap_token_lp1 = view!(
```

Comment:
lets rename `before_swap` to `before_withdraw`. same for `after_`.

Reply:
ok